### PR TITLE
Adding parallel make_heap 

### PR DESCRIPTION
**Pull request by hkaiser**
This revives #1914, which has never made it in. This also converts the algorithm to CPOs and makes it conforming to C++20. Thanks again @Syntaf for working on this and sorry it took so long to get back to it.

### CODE REVIEW

**Comment by Syntaf** [comment]
Very cool to see this being revisited!

**Review by msimberg — APPROVED** [review]
Very nice, thanks!

File: libs/include/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2020 The STE||AR-Group
```
[the remark this answers is not in the document]
Comment (msimberg) [reply libs/include/CMakeLists.txt]: Leftover comment from a previously started review, please ignore.

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/make_heap.hpp [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+                    policy.parameters(), policy.executor());
+
+                std::vector<hpx::future<void>> workitems;
+                std::list<std::exception_ptr> errors;
+
+                using tuple_type = hpx::util::tuple<RndIter, std::size_t>;
+
+                auto op = [=](tuple_type const& t) {
+                    sift_down_range(first, comp, proj, (std::size_t) n,
+                        hpx::util::get<0>(t), hpx::util::get<1>(t));
```
Comment (msimberg) [inline libs/parallelism/algorithms/include/hpx/parallel/algorithms/make_heap.hpp:280]:
```suggestion
                using tuple_type = hpx::tuple<RndIter, std::size_t>;

                auto op = [=](tuple_type const& t) {
                    sift_down_range(first, comp, proj, (std::size_t) n,
                        hpx::get<0>(t), hpx::get<1>(t));
```
Looks like the include with `hpx/modules/datastructures.hpp` or `hpx/datastructures/tuple.hpp` is missing. I also noticed yesterday that the deprecation warnings were not enabled for 1.6.0, and I've started updating names that I missed in the previous deprecation PRs (there are quite a few, so I recommend you don't enable it here).

**Review by msimberg — CHANGES_REQUESTED** [review]
I take back my previous review. It's just some names that need to be updated though.

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/make_heap.hpp (hunk 2 above)
Comment (hkaiser) [reply libs/parallelism/algorithms/include/hpx/parallel/algorithms/make_heap.hpp:280]: Correct, I'll add the missing header file. Thanks for noticing!

File: libs/include/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (hkaiser) [reply libs/include/CMakeLists.txt]: Yes, this was a leftover from the rebase. Should be removed now.

**Comment by msimberg** [comment]
Hmm, this is maybe not rebased on master? I think the CircleCI failure is just because of that. The jenkins builders build a merge with master.

**Comment by hkaiser** [comment]
> Hmm, this is maybe not rebased on master? I think the CircleCI failure is just because of that. The jenkins builders build a merge with master.

I rebased this again, let's see.

**Review by msimberg — APPROVED** [review]
Thanks @hkaiser, looks good now!

**Pull request merged**